### PR TITLE
clfft 2.10.0

### PR DESCRIPTION
**Pull request by Moisan**
As pointed in #3254, clfft needed to be bumped. I removed the patch that has been merged upstream in this release.

### Have you:

- [X] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [X] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [X] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [X] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [X] Removed the `revision` line, if any, when bumping the version number?
- [X] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [X] Not altered the `bottle` section?
- [X] Checked that the tests still pass?

### CODE REVIEW

**Comment by dpo** [comment]
Many thanks. Something's not quite right with the test:
```
==> /usr/bin/clang++ /usr/local/Cellar/clfft/2.10.0/share/clfft/examples/fft1d.c -I/usr/local/opt/clfft/include -L/usr/local/opt/clfft/lib -lclFFT -framework OpenCL -o fft1d
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated
Error: homebrew/science/clfft: failed
<"Platform found: Apple\nDevice found on the above platform: Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz\n\nPerforming fft on an one dimensional array of size N = 16\n(0.000000, 0.000000) (1.000000, 3.000000) (2.000000, 6.000000) (3.000000, 9.000000) (4.000000, 12.000000) (5.000000, 15.000000) (6.000000, 18.000000) (7.000000, 21.000000) (8.000000, 24.000000) (9.000000, 27.000000) (10.000000, 30.000000) (11.000000, 33.000000) (12.000000, 36.000000) (13.000000, 39.000000) (14.000000, 42.000000) (15.000000, 45.000000) \n\nfft result: \n(0.000000, 0.000000) (1.000000, 3.000000) (2.000000, 6.000000) (3.000000, 9.000000) (4.000000, 12.000000) (5.000000, 15.000000) (6.000000, 18.000000) (7.000000, 21.000000) (8.000000, 24.000000) (9.000000, 27.000000) (10.000000, 30.000000) (11.000000, 33.000000) (12.000000, 36.000000) (13.000000, 39.000000) (14.000000, 42.000000) (15.000000, 45.000000) \n"> expected to be =~
</\(120\.000000,\ 360\.000000\)/>.
```

File: clfft.rb [hunk 1]
```diff
@@ -31,39 +28,3 @@ def install
     assert_match "(120.000000, 360.000000)", `./fft1d`
```
Comment (dpo) [inline clfft.rb]: `ENV.cxx` should be `ENV.cc`. Also the result of the test isn't as expected.

Comment (Moisan) [inline clfft.rb]: I get following error when using `ENV.cc`:

```
Undefined symbols for architecture x86_64:
  "_clfftInitSetupData", referenced from:
      _main in fft1d-7920e7.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: homebrew/science/clfft: failed
```

So either there is some dependency missing for the test or I am missing something.

Also the test passes locally with `ENV.cxx`, it seems to be confused by the architecture used by the jenkins server.

**Comment by dpo** [comment]
All good thank you.

**Pull request closed without merging**